### PR TITLE
feat: Add Scroll fee vault address as fee recipient

### PR DESCRIPTION
**Pull request by Thegaram**
`FeeVaultAddress` can be configured in genesis. If not configured, then we fall back to normal Clique operation (signer gets fees).

You can see an example fee vault implementation [here](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/universal/FeeVault.sol). We could pre-deploy it in genesis or simply use a "single use transaction" (that results in a predictable address).

### CODE REVIEW

File: core/blockchain_test.go [hunk 1]
```diff
@@ -2973,10 +2977,7 @@ func TestEIP1559Transition(t *testing.T) {
 
 	// 6+5: Ensure that miner received only the tx's effective tip.
```
Comment (0xmountaintop) [inline core/blockchain_test.go]: should we update the comments here?

**Review by zimpha — DISMISSED** [review]

File: params/config.go [hunk 2]
```diff
@@ -258,16 +258,16 @@ var (
 	//
 	// This configuration is intentionally not using keyed fields to force anyone
 	// adding flags to the config to also have to set these fields.
-	AllEthashProtocolChanges = &ChainConfig{big.NewInt(1337), big.NewInt(0), nil, false, big.NewInt(0), common.Hash{}, big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), nil, new(EthashConfig), nil, false}
+	AllEthashProtocolChanges = &ChainConfig{big.NewInt(1337), big.NewInt(0), nil, false, big.NewInt(0), common.Hash{}, big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), nil, new(EthashConfig), nil, false, &common.Address{123}}
```
Comment (icemelon) [inline params/config.go]: should we use `nil` for `FeeVaultAddress`? For example, we also set `Zktrie` to false here.

File: params/config.go [hunk 3]
```diff
@@ -258,16 +258,16 @@ var (
...
 
 	// AllCliqueProtocolChanges contains every protocol change (EIPs) introduced
 	// and accepted by the Ethereum core developers into the Clique consensus.
 	//
 	// This configuration is intentionally not using keyed fields to force anyone
 	// adding flags to the config to also have to set these fields.
-	AllCliqueProtocolChanges = &ChainConfig{big.NewInt(1337), big.NewInt(0), nil, false, big.NewInt(0), common.Hash{}, big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), nil, nil, nil, &CliqueConfig{Period: 0, Epoch: 30000}, false}
+	AllCliqueProtocolChanges = &ChainConfig{big.NewInt(1337), big.NewInt(0), nil, false, big.NewInt(0), common.Hash{}, big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), nil, nil, nil, &CliqueConfig{Period: 0, Epoch: 30000}, false, &common.Address{123}}
```
Comment (icemelon) [inline params/config.go]: ditto

**Review by zimpha — APPROVED** [review]

**Review by 0xmountaintop — APPROVED** [review]

**Comment by 0xmountaintop** [comment]
@icemelon would you like to give 1 more review on this?

**Review by icemelon — APPROVED** [review]

**Comment by Thegaram** [comment]
Note: Not marking this as a `breaking-change` since it's backward compatible if we don't change the config.

**Comment by iczc** [comment]
> Note: Not marking this as a `breaking-change` since it's backward compatible if we don't change the config.

There is a situation where, if the network genesis is updated to include the FeeVaultAddress config and node A uses the new l2geth with custom fee recipient, while Node B uses the old l2geth as well as still using the coinbase fee recipient due to not supporting new config, they will experience consensus failure. 

However, during the testnet stage, as all nodes are under our control and upgraded together, the issue will not arise yet.

**Pull request merged**